sqlglot/dialects/singlestore.py

```diff
@@ -290,6 +290,25 @@ class Parser(MySQL.Parser):
             "USERS FOR GROUP": _show_parser("USERS FOR GROUP", target=True),
         }
 
+        def _parse_vector_expressions(
+            self, expressions: t.List[exp.Expression]
+        ) -> t.List[exp.Expression]:
+            type_name = expressions[1].name.upper()
+            if type_name == "I8":
+                type_name = "TINYINT"
+            elif type_name == "I16":
+                type_name = "SMALLINT"
+            elif type_name == "I32":
+                type_name = "INT"
+            elif type_name == "I64":
+                type_name = "BIGINT"
+            elif type_name == "F32":
+                type_name = "FLOAT"
+            elif type_name == "F64":
+                type_name = "DOUBLE"
```

Comment:
Nit: can we factor these into a mapping `VECTOR_TYPE_ALIASES` write it a bit more compactly? We should be able to leverage this in `datatype_sql` too.